lib/model/Channel.js

```diff
@@ -0,0 +1,117 @@
+const path = require('path');
+
+const Capability = require(path.join(__dirname, 'Capability.js'));
+const FineChannel = require(path.join(__dirname, 'FineChannel.js'));
+const SwitchingChannel = require(path.join(__dirname, 'SwitchingChannel.js'));
+
+module.exports = class Channel {
+  constructor(jsonObject, key) {
+    this.jsonObject = jsonObject; // calls the setter
+    this._key = key;
+  }
+
+  set jsonObject(jsonObject) {
+    this._jsonObject = jsonObject;
+    this._cache = {};
+  }
+
+  get key() {
+    return this._key;
+  }
+
+  get name() {
+    return this._jsonObject.name || this.key;
+  }
+
+  get type() {
+    return this._jsonObject.type; // required
+  }
+
+  get color() {
+    return this._jsonObject.color || null; // required if SingleColor, else forbidden
+  }
+
+  // ordered by fineness (coarsest first)
+  // -> that makes it different from calling Object.keys(ch.fineChannels);
+  get fineChannelAliases() {
+    return this._jsonObject.fineChannelAliases || [];
+  }
+
+  // { 'fine channel alias': FineChannel, ... }
```

Comment:
Why not just an array? We can get the key from a FineChannel instance and we can get an array of all keys via `fineChannelAliases`.